ttnn/cpp/ttnn/operations/data_movement/repeat/repeat.cpp

```diff
@@ -18,10 +21,18 @@ ttnn::Tensor RepeatOperation::invoke(
     const Shape & repeat_dims,
     const std::optional<MemoryConfig>& memory_config_arg) {
 
-    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    // special path for handling padded tiled tensors that are naively repeated
+    // *with padding* in the current implementation
+    std::optional<Tensor> formatted_input_tensor = std::nullopt;
+    if (input_tensor.get_layout() != Layout::ROW_MAJOR
+        && input_tensor.get_legacy_shape().without_padding() != input_tensor.get_legacy_shape()) {
```

Comment:
Could you please reuse new methods that became available in
https://github.com/tenstorrent/tt-metal/pull/13372? `input_tensor.get_padded_shape() != input_tensor.get_logical_shape()`

Reply:
Yes, didn't realize those had landed. Exciting! :)